app/graphql/corpus_recommendation.py

```diff
@@ -8,5 +8,7 @@
 @strawberry.experimental.pydantic.type(model=CorpusRecommendationModel)
 class CorpusRecommendation:
     id: strawberry.ID
+    tile_id: strawberry.auto = strawberry.field(
```

Comment:
it looks like tile_id is something very specific to new tab, why are we trying to reuse the Home domain model here instead of creating a new one?

Reply:
`CorpusRecommendation` is not necessarily tied to Home, but can be used for any recommendation made from the Corpus.

That said, you have a great point that Firefox is the only client that would need a `tile_id`, and it would be cleaner to have a separate type for this. :writing_hand: Let me take a stab at that..

Reply:
After going down the path of separating types and reflecting on this, I lean towards keeping `tile_id` in `CorpusRecommendation`:
- Adding a `NewTabCorpusRecommendation` type also requires a separate `NewTabCorpusSlate` type, a separate `NewTabCorpusRecommendation` resolver, etc. Additional changes might also be required in Snowplow JSON schema and Dbt modeling.
- [DIS-491](https://getpocket.atlassian.net/browse/DIS-491) `tileId` is deprecated and I am going to work with Firefox Devs and Firefox Data engineering to replace it with `CorpusRecommendation.id`. This will bring Firefox New Tab in line with how telemetry is identified on Home.
- One of the advantages of GraphQL is that clients can have great freedom in querying exactly the information they need. My first thought was that Firefox would not benefit from any extensions to `CorpsuRecommendations` because its product requirements are set in stone. But maybe it's also the other way around, in that New Tab product requirements have not changed much because the existing Rest API was so hard to change?

@eu9ene  Are you ok with keeping the existing strategy of using a deprecation message to prevent any clients except Firefox New Tab from using `tileId`, and removing this field over time?

Reply:
Yeah, unification and having the same pipelines for all recs seems like a good strategy too. Let's add the depreciation flag and maybe some other attributes or comments to let the clients know that some fields are relevant only for specific scenarios. My concern here is that when deployed, this API will be hard to change, so we should plan for the future.

Reply:
:+1: I expanded the deprecation reason on the line below to this:

> Only to be used by Firefox. Other clients should use `id`. We plan to also migrate Firefox New Tab to use CorpusRecommendation.id instead of tileId to track recommendation telemetry.

I appreciate your focus on the API. In hindsight I should've included you for review on the [API proposal](https://getpocket.atlassian.net/wiki/spaces/PE/pages/2915368961/Pocket+New+Tab+Recommendations+for+New+Markets). I'll try to make sure to remember to include you in the future.

Reply:
That's why it might be a good idea to do those proposals as a part of a pull request. They might look fine in confluence but be inconsistent with the rest of the code.